cmake/modules/SwiftConfigureSDK.cmake

```diff
@@ -335,6 +337,15 @@ macro(configure_sdk_unix name architectures)
     set(SWIFT_SDK_${prefix}_STATIC_ONLY FALSE)
   endif()
 
+  if("${prefix}" STREQUAL "LINUX"
```

Comment:
Uninformed question: how likely it is that a linker for these platform does not support `--build-id`?
I'm seeing that all of BFD, gold, lld and mold support this flag, nevertheless I would be tempted to have an option for the user to influence this.

Reply:
I don't think any of the extant ELF linkers lack support, and since I think it's a feature almost everyone wants to turn on (especially Linux distros themselves, so they can split out debug symbols), the chances of a new linker not supporting it are basically zero.